### PR TITLE
feat: Enhance InMemoryCache 'get' method

### PR DESCRIPTION
**Pull request by IvayloAndonov**
# Context

In case when the cache is cleared and multiple simultaneous requests come in, we would run the DB query multiple times.
Raised by @mkazlauskas 

# Proposed Solution

Caching the promise instead of resolving the value, so the next invocation will instantly return the same promise.
In case of DB query fails -> `result.catch()` resets the cache so don't have to wait for the TTL

# Important Changes Introduced

### CODE REVIEW

**Review by rhyslbw — CHANGES_REQUESTED** [review]
It would be best to have a test demonstrate this

File: packages/cardano-services/src/InMemoryCache/InMemoryCache.ts
Comment (rhyslbw) [inline packages/cardano-services/src/InMemoryCache/InMemoryCache.ts:30]:
```suggestion
    const resultPromise = dbSyncQuery();
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]
What about the test scenario of multiple requests being received before the cache is primed?

File: packages/cardano-services/src/InMemoryCache/InMemoryCache.ts
Comment (IvayloAndonov) [reply packages/cardano-services/src/InMemoryCache/InMemoryCache.ts]: [52a24a2](https://github.com/input-output-hk/cardano-js-sdk/pull/286/commits/52a24a29a6bc040355e2da5a676a9096e50fe987)

**Review by mkazlauskas — APPROVED** [review]
Looks good! The only improvement that comes to mind is a test that `.catch` resets cache, but I think it's good enough as-is.

Also, `perf` seems like a good commit type for this

**Comment by IvayloAndonov** [comment]
> Looks good! The only improvement that comes to mind is a test that `.catch` resets cache, but I think it's good enough as-is.
> 
> Also, `perf` seems like a better commit type for this

Yeah, I'll squash the commits as `perf` once I've got another approval and its ready for merge

**Review by rhyslbw — DISMISSED** [review]

**Review by mkazlauskas — APPROVED** [review]

**Pull request merged**
